Clover/app/src/main/java/org/floens/chan/ui/controller/FiltersController.java

```diff
@@ -130,6 +136,33 @@ public void onCreate() {
     public void onClick(View v) {
         if (v == add) {
             showFilterDialog(new Filter());
+        } else if (v == enable) {
+            //if every filter is disabled, enable all of them and set the drawable to be an x
+            //if every filter is enabled, disable all of them and set the drawable to be a checkmark
+            //if some filters are enabled, disable them and set the drawable to be a checkmark
+            List<Filter> enabledFilters = filterEngine.getEnabledFilters();
+            List<Filter> allFilters = filterEngine.getAllFilters();
+            if(enabledFilters.isEmpty()) {
+                for (Filter filter : allFilters) {
```

Comment:
do you think it would be beneficial to wrap and paramaterize this loop, to reduce duplicate code between the conditionals?
```
if (enabledFilters.isEmpty()) {
  setFiltersEnabled(allFilters, true)
  setDrawableTo(R.drawable.ic_clear_white_24dp, v)
} else if...
```
where `setFilters(List<Filter> filters, boolean enabled)` will set the enabled attribute to the passed in parameter

Reply:
That's a good idea, I'll implement that.